src/util.ts

```diff
@@ -643,3 +645,71 @@ export function intHash(str : string, length = 10) : number {
     return (hash % length + length) % length; // Ensure the result is non-negative
 }
 
+/**
+ * Evaluate a JSON query expression against the provided data.
+ * @param data The data to evaluate the JSON query against.
+ * @param jsonPath The JSON path or custom JSON query expression.
+ * @param jsonPathOperator The operator to use for comparison.
+ * @param expectedValue The expected value to compare against.
+ * @returns An object containing the status and the evaluation result.
+ * @throws Error if the evaluation returns undefined.
+ */
+export async function evaluateJsonQuery(data: any, jsonPath: string, jsonPathOperator: string, expectedValue: any): Promise<{ status: boolean; response: any }> {
+    // Attempt to parse data as JSON; if unsuccessful, handle based on data type.
+    let response: any;
+    try {
+        response = JSON.parse(data);
+    } catch {
+        response = typeof data === "number" || typeof data === "object" ? data : data.toString();
+    }
+
+    try {
+        // If a JSON path is provided, pre-evaluate the data using it.
+        response = (jsonPath) ? await jsonata(jsonPath).evaluate(response) : response;
+
+        if (typeof response === "object" || Array.isArray(response) || response instanceof Date || typeof response === "function") {
+            throw new Error(`The post-JSON query evaluated response from the server is of type ${typeof response}, which cannot be directly compared to the expected value`);
```

Comment:
In my testing, this error throw is falsely catching type `Buffer` (OctetString) from SNMP responses which jsonata can handle readily as strings. I need to re-tool this, or remove it entirely before this PR can ship.

Reply:
Resolved with 8d8ce23